packages/@coorpacademy-app-player/src/view/state-to-props/test/player.js

```diff
@@ -138,3 +140,102 @@ test('should display "Go to question" for the context tab cta', t => {
   const props = playerProps(state);
   t.is(props.cta.submitValue, 'Go to question');
 });
+
+test('should display showNewMedia notification for the answer route', t => {
+  const state = {
+    data: {
+      progressions: {
+        entities: {
+          basic: createProgression(basicSlide)
+        }
+      }
+    },
+    ui: {
+      current: {progressionId: 'basic'},
+      route: {foo: 'answer'}
+    }
+  };
+
+  const props = playerProps(state);
+  t.is(props.showNewMedia, true);
+});
+
+test('should not display showNewMedia notification for the answer route', t => {
+  let state = {
+    data: {
+      progressions: {
+        entities: {
+          basic: createProgression(basicSlide)
+        }
+      }
+    },
+    ui: {
+      current: {progressionId: 'basic'},
+      route: {foo: 'answer'}
+    }
+  };
+
+  state = set('data.progressions.entities.basic.state.viewedResources', ['videoRef'], state);
+
+  const props = playerProps(state);
+  t.is(props.showNewMedia, false);
+});
+
+test('should display showNewMedia notification for the undefined route', t => {
+  const state = {
+    data: {
+      progressions: {
+        entities: {
+          basic: createProgression(basicSlide)
+        }
+      }
+    },
+    ui: {
+      current: {progressionId: 'basic'},
+      route: {foo: undefined}
+    }
+  };
+
+  const props = playerProps(state);
+  t.is(props.showNewMedia, true);
+});
+
+test('should not display showNewMedia notification for the undefined route', t => {
```

Comment:
`should not display showNewMedia` --> `should not display new media notification`


Ce serait bien que dans le titre du test, on comprenne dans quel cas ce que tu annonces arrive.

Si je lis `should display showNewMedia notification for the undefined route`, je m'attends à ce que `showNewMedia` soit toujours vrai lorsqu'on est sur la route `undefined`.
Le test suivant `should display showNewMedia notification for the undefined route` du coup contredit ça dans ma tête.
Si tu rajoute toutes les conditions en rapport avec ce que tu tests, ça devient plus clair (en rajoutant dans le cas de ce test-ci " when the media has been seen" par exemple, et l'inverse pour les autres).
 
Pas clair : "should do X" et "should not do X"
Plus clair : "should do X when Y" et "should not do X when Z"


Pour que ça soit plus simple, tu pourrais faire un premier test comme celui-ci, qui teste tous les cas où le média est déjà vu.

```js
test('should not display new media notification when user has seen the media', t => {
  const state = {
    data: {
      progressions: {
        entities: {
          basic: set('state.viewedResources', ['videoRef'], createProgression(basicSlide))
        }
      }
    },
    ui: {
      current: {progressionId: 'basic'},
      route: {foo: undefined}
    }
  };

  [undefined, 'clue', 'media', 'coach', 'answer'].forEach(route => {
    const props = playerProps(state);
    t.false(props.showNewMedia);
  });
});
```

Ensuite, tu rajoutes/renomme des tests du style `should <not> display new media notification when route is ...`